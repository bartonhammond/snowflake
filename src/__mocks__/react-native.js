/**
 * # __mockes__/react-native.js
 * 
 * This class stubs out the React-Native classes with React classes
 */
'use strict';
/**
 * ## Imports
 * 
 * ReactNative is actually React
 */ 
const React = require('react');
const ReactNative = React;

/**
 * ## These need additional mocking
 * 
 * ReactNative is actually React
 */ 
ReactNative.StyleSheet = {
    create: function create(styles) {
        return styles;
    }
};
class View extends React.Component {
    render() { return false; }
}
class PixelRatio extends React.Component {
    static get() { return 1; }
}
/**
 * ## Stubs
 * 
 * Simple replacements for testing
 */ 
ReactNative.View = View;
ReactNative.ScrollView = View;
ReactNative.Text = View;
ReactNative.TouchableOpacity = View;
ReactNative.TouchableHighlight = View;
ReactNative.TouchableWithoutFeedback = View;
ReactNative.ToolbarAndroid = View;
ReactNative.Image = View;
ReactNative.PixelRatio = PixelRatio;
ReactNative.NativeModules= {};

ReactNative.Platform = {};
module.exports = ReactNative;
