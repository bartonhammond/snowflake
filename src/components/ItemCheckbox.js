/**
 * # ItemCheckbox.js
 *
 * This class was initially written by
 * https://github.com/mhollweck/react-native-item-checkbox
 *
 * I've opened an issue to attempt to merge this back in
 */
'use strict'

/**
 * ## Imports
 *
 * React
 */
import React, {PropTypes} from 'react'
import
{
  View,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback
} from 'react-native'

 /**
  * The vector icon
  */
import Icon from 'react-native-vector-icons/FontAwesome'

var ItemCheckbox = React.createClass({
  /**
   * ## ItemCheckbox class
   *
   * set the propTypes
   */
  propTypes: {
    onCheck: PropTypes.func,
    onUncheck: PropTypes.func,
    icon_check: PropTypes.string,
    icon_open: PropTypes.string,
    size: PropTypes.number,
    backgroundColor: PropTypes.string,
    color: PropTypes.string,
    iconSize: PropTypes.string,
    checked: PropTypes.bool,
    style: PropTypes.func,
    text: PropTypes.string,
    disabled: PropTypes.bool
  },
  /**
   * ### getDefaultProps
   * set the default values
   */
  getDefaultProps: function () {
    return {
      onCheck: null,
      onUncheck: null,
      icon_check: 'check-square',
      icon_open: 'square-o',
      size: 30,
      backgroundColor: 'white',
      color: 'grey',
      iconSize: 'normal',
      checked: false,
      text: 'MISSING TEXT',
      disabled: false
    }
  },
  /**
   * ### getInitialState
   *
   * Set the box to be checked or not
   */
  getInitialState: function () {
    return {
      checked: this.props.checked,
      bg_color: this.props.backgroundColor
    }
  },
  /**
   * ### _getCircleCheckSytel
   * merge the props styles w/ some defaults
   */
  _getCircleCheckStyle: function () {
    return {
      width: this.props.size,
      height: this.props.size,
      backgroundColor: this.state.bg_color,
      borderColor: this.props.color,
      borderWidth: 2,
      borderRadius: this.props.size / 2,
      justifyContent: 'center',
      alignItems: 'center',
      padding: 2
    }
  },
  /**
   * ### _completeProgress
   * If the checkbox is pressable, figure out what state it's in and
   * what the display should look like
   */
  _completeProgress: function () {
    if (this.state.checked) {
      this.setState({
        checked: false,
        bg_color: this.props.backgroundColor
      })
      if (this.props.onUncheck) {
        this.props.onUncheck()
      }
    } else {
      this.setState({
        checked: true,
        bg_color: this.props.color
      })
      if (this.props.onCheck) {
        this.props.onCheck()
      }
    }
  },
  /**
   * ### componentDidMount
   * If there is a ```checked``` property, set the UI appropriately
   */
  componentDidMount: function () {
    if (this.props.checked) {
      this._completeProgress()
    }
  },
  /**
   * ### render
   * Use Touchable with or without Feedback depending on
   * ```disabled```.
   * Set the ```iconName``` depending on if checked
   */
  render: function () {
    var iconName = this.props.icon_open
    if (this.state.checked) {
      iconName = this.props.icon_check
    }
    if (this.props.disabled) {
      iconName = this.props.checked ? this.props.icon_check : this.props.icon_open
      return (
        <View style={this.props.style}>
          <TouchableWithoutFeedback>
            <View style={{
              flexDirection: 'row',
              flex: 1
            }}>
              <Icon
                name={iconName}
                size={20}
              />
              <Text> {this.props.text}</Text>
            </View>
          </TouchableWithoutFeedback>
        </View>
        )
    } else {
      return (
        <View style={this.props.style}>
          <TouchableHighlight
            onPress={this._completeProgress} >
            <View style={{
              flexDirection: 'row',
              flex: 1
            }}>
              <Icon
                name={iconName}
                size={20} />
              <Text> {this.props.text}</Text>
            </View>
          </TouchableHighlight>
        </View>
        )
    }
  }
})

module.exports = ItemCheckbox
