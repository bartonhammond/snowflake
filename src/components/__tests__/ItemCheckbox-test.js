/**
 * # ItemCheckbox-test.js
 *
 * This class tests that the ItemCheckbox renders correctly under
 * numerous conditions
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 *
 */
'use strict'

import 'react-native'
import React from 'react'

import ItemCheckbox from '../ItemCheckbox'

import ReactTestUtils from 'react-addons-test-utils'
const renderer = ReactTestUtils.createRenderer()

/**
 * ## Test
 */
describe('ItemCheckbox', () => {
  /**
   * ### if not disabled and checked, it should display check-square and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if not disabled and checked, it should display check-square and text', () => {
    const props = {
      checked: true,
      text: 'TextShouldDisplay',
      disabled: false
    }
    renderer.render(<ItemCheckbox {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if not disabled and not checked, it should display square-o and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if not disabled and not checked, it should display square-o and text', () => {
    const props = {
      checked: false,
      text: 'TextShouldDisplay',
      disabled: false
    }
    renderer.render(<ItemCheckbox {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if disabled and checked, it should display check-square and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if disabled and checked, it should display check-square and text', () => {
    const props = {
      checked: true,
      text: 'TextShouldDisplay',
      disabled: true
    }
    renderer.render(<ItemCheckbox {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if disabled and not checked, it should display square-o and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if disabled and not checked, it should display square-o and text', () => {
    const props = {
      checked: false,
      text: 'TextShouldDisplay',
      disabled: true
    }
    renderer.render(<ItemCheckbox {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})// describe ItemCheckbox

