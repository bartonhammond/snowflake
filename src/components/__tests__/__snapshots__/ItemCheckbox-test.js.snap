exports[`ItemCheckbox if disabled and checked, it should display check-square and text 1`] = `
<View
  style={undefined}>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Icon
        allowFontScaling={false}
        name="check-square"
        size={20} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ItemCheckbox if disabled and not checked, it should display square-o and text 1`] = `
<View
  style={undefined}>
  <TouchableWithoutFeedback>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Icon
        allowFontScaling={false}
        name="square-o"
        size={20} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ItemCheckbox if not disabled and checked, it should display check-square and text 1`] = `
<View
  style={undefined}>
  <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    underlayColor="black">
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Icon
        allowFontScaling={false}
        name="check-square"
        size={20} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;

exports[`ItemCheckbox if not disabled and not checked, it should display square-o and text 1`] = `
<View
  style={undefined}>
  <TouchableHighlight
    activeOpacity={0.85}
    onPress={[Function]}
    underlayColor="black">
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Icon
        allowFontScaling={false}
        name="check-square"
        size={20} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </TouchableHighlight>
</View>
`;
